test/service/actions/posts.test.js

```diff
@@ -189,30 +193,35 @@ describe('Actions.Posts', () => {
                 TestHelper.fakePost(channelId)
             );
 
+            let shouldStop = false;
             store.subscribe(() => {
                 const state = store.getState();
                 const getRequest = state.requests.posts.getPost;
                 const {posts, postsByChannel} = state.entities.posts;
 
-                if (getRequest.status === RequestStatus.SUCCESS) {
-                    assert.ok(posts);
-                    assert.ok(postsByChannel);
-                    assert.ok(postsByChannel[channelId]);
-
-                    assert.ok(posts[post.id]);
-
-                    let found = false;
-                    for (const postIdInChannel of postsByChannel[channelId]) {
-                        if (postIdInChannel === post.id) {
-                            found = true;
-                            break;
+                if (!shouldStop &&
+                    (getRequest.status === RequestStatus.SUCCESS || getRequest.status === RequestStatus.FAILURE)) {
```

Comment:
Maybe we should discuss a standard for this since I keep changing these to
```javascript
if (request.status === SUCCESS) {
    // request.error should be falsey here
    // ASSERTS
    done();
} else if (request.status === FAILURE) {
    // request.error should be truthy here
    done(request.error);
}
```
and you're now changing it back 😛 

My reasoning for how I do it is that if the request was successful, then error will always be false and if it failed, then error will always be set. As long as those assumptions are always true, we can remove one layer of nesting here to clean up the code a bit.